test/lib/jdk/test/lib/Platform.java

```diff
@@ -362,6 +362,18 @@ public static String sharedLibraryExt() {
         }
     }
 
+    /**
+     * Returns the usual file prefix of a shared library, e.g. "lib" on linux, empty on windows.
+     * @return file name prefix
+     */
+    public static String sharedLibraryPrefix() {
+        if (isWindows()) {
+            return "";
+        } else {
+            return "lib";
+        }
+    }
+
```

Comment:
The following tests could leverage this new API. Just look for calls to `Platform.sharedLibraryExt()`:

test/hotspot/jtreg/runtime/signal/SigTestDriver.java
test/hotspot/jtreg/vmTestbase/nsk/jvmti/NativeLibraryCopier.java
test/jdk/com/sun/tools/attach/warnings/DynamicLoadWarningTest.java

Perhaps a `Platform.buildSharedLibraryName()` API is worth considering.

Reply:
Agreed with suggestion from Chris.

Reply:
Agreed, I'd like to see `Platform.buildSharedLibraryName()` in addition to the other methods (the later are needed if you want to decompose a filename to get the library name (though we could also provide a function just to do that).